docs/basic/complexity.md

```diff
@@ -1,60 +1,155 @@
+2. Average-case time complexity, which is the average of the complexity on every possible inputs of a given size for every scale of data scale. (Or, the expected complexity on random input.)
 
-同样的，我们使用 $\Omega$ 符号来描述一个函数的渐进下界。$f(n)=\Omega(g(n))$，当且仅当 $\exists c,n_0$，使得 $\forall n \ge n_0,0\le c\cdot g(n)\le f(n)$。
+As "the trend of time cost's growing with data scale's increasing" is an ambiguous concept, we need to represent the time complexity formally using **asymptotic notation** introduced below.
 
-### 小 o 符号
+## Asymptotic Notation
 
-如果说 $O$ 符号相当于小于等于号，那么 $o$ 符号就相当于小于号。
+Briefly, the asymptotic notation ignored the slower growing part of a function and its coefficient and constants, but preserved the important part representing the growing trend of this function. <!---这部分过于math需要指导-->
 
-$f(n)=o(g(n))$，当且仅当对于任意给定的正数 $c$，$\exists n_0$，使得 $\forall n \ge n_0,0\le f(n)< c\cdot g(n)$。
+### Θ-Notation
 
-### 小 ω 符号
+For function $f(n), g(n)$, 
+$$ f(n) = \Theta(g(n))$$ 
+If and only if:
+$$\exists c_1,c_2,n_0>0\colon \forall n \ge n_0, 0\le c_1\cdot g(n)\le f(n) \le c_2\cdot g(n)$$
 
-如果说 $\Omega$ 符号相当于大于等于号，那么 $\omega$ 符号就相当于大于号。
+Which is, in plain language, if $f(n) = \Theta(g(n))$, then we can find two positive numbers $c_1, c_2$, <!---不会了，ksyx救我-->
 
-$f(n)=\omega(g(n))$，当且仅当对于任意给定的正数 $c$，$\exists n_0$，使得 $\forall n \ge n_0,0\le c\cdot g(n)< f(n)$。
+For example $3n^2+5n-3=\Theta(n^2)$,$n\sqrt n + n\log^5 n+m\log m+nm=\Theta(n\sqrt n+m\log m+nm)$.
+
+### Big O Notation
+
+Θ-Notation gives us upper and lower bound of a function in the same time.<!---Wondering for better expression--> We can use big O notation<!---is it necessary to use math expression?(the same below)--> if we only know its asymptotical upper bound but not lower, which is
+$$f(n)=O(g(n))$$, 
+if and only if 
+$$\exists c,n_0\colon \forall n \ge n_0,0\le f(n)\le c\cdot g(n)$$
+
+It is more common to use big O notation while researching time complexity, as we are usually concerned with the upper bound of the program's time cost rather than the lower bound.
+
+Note that the "upper" and "lower" bounds here are for the trend of the function, not for the algorithm, whose upper case refers to worst-case time complexity but not big O notation. Therefore, using Θ-notation is completely viable to represent worst-case time complexity, and it can even be said that Θ-notation is more accurate than big O notation. As for the main reason for using O notation more often, one is we sometimes can prove the upper bound rather than lower bound (this is usually the case for more complicated algorithm and its complexity analysis), and the other is O is easier to input in a computer. 
```

Comment:
```suggestion
Note that the "upper" and "lower" bounds here are for the trend of the function, not for the algorithm, whose upper case refers to worst-case time complexity but not big O notation. Therefore, using Θ-notation is completely viable to represent worst-case time complexity, and it can even be said that Θ-notation is more accurate than big O notation. As for the main reason for using O notation more often, one is we sometimes can prove the upper bound, but not lower bound (this is usually the case for more complicated algorithm and analyzing its complexity), and the other reason is the character O is easier to input in a computer. 
```